proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1775 @@
+
+*([Issue](https://github.com/sass/sass/issues/2831))*
+
+This proposal adds Sass support for several new CSS color spaces defined in
+[CSS Color Level 4][color-4], including access to non-RGB color models and
+colors outside the sRGB gamut.
+
+[color-4]: https://www.w3.org/TR/css-color-4/
+
+## Table of Contents
+
+* [Summary](#summary)
+  * [Design Decisions](#design-decisions)
+* [Definitions](#definitions)
+  * [Color](#color)
+  * [Legacy Color](#legacy-color)
+  * [Known Color Space](#known-color-space)
+  * [Predefined Color Spaces](#predefined-color-spaces)
+  * [Missing Components](#missing-components)
+  * [Powerless Components](#powerless-components)
+  * [Color Interpolation Method](#color-interpolation-method)
+* [Procedures](#procedures)
+  * [Converting a Color](#converting-a-color)
+  * [Gamut Mapping](#gamut-mapping)
+  * [Parsing Color Components](#parsing-color-components)
+  * [Percent-Converting a Number](#percent-converting-a-number)
+  * [Normalizing Color Channels](#normalizing-color-channels)
+  * [Interpolating Colors](#interpolating-colors)
+    * [Premultiply Transparent Colors](#premultiply-transparent-colors)
+    * [Hue Interpolation](#hue-interpolation)
+* [Deprecated Functions](#deprecated-functions)
+* [New Color Module Functions](#new-color-module-functions)
+  * [`color.space()`](#colorspace)
+  * [`color.to-space()`](#colorto-space)
+  * [`color.is-legacy()`](#coloris-legacy)
+  * [`color.is-powerless()`](#coloris-powerless)
+  * [`color.is-in-gamut()`](#coloris-in-gamut)
+  * [`color.to-gamut()`](#colorto-gamut)
+  * [`color.channel()`](#colorchannel)
+* [Modified Color Module Functions](#modified-color-module-functions)
+  * [`color.hwb()`](#colorhwb)
+  * [`color.mix()`](#colormix)
+  * [`color.change()`](#colorchange)
+  * [`color.adjust()`](#coloradjust)
+  * [`color.scale()`](#colorscale)
+  * [`color.complement()`](#colorcomplement)
+  * [`color.invert()`](#colorinvert)
+  * [`color.grayscale()`](#colorgrayscale)
+  * [`color.ie-hex-str()`](#colorie-hex-str)
+* [New Global Functions](#new-global-functions)
+  * [`hwb()`](#hwb)
+  * [`lab()`](#lab)
+  * [`lch()`](#lch)
+  * [`oklab()`](#oklab)
+  * [`oklch()`](#oklch)
+  * [`color()`](#color)
+* [Modified Global Functions](#modified-global-functions)
+  * [`rgb()` and `rgba()`](#rgb-and-rgba)
+  * [`hsl()` and `hsla()`](#hsl-and-hsla)
+
+## Background
+
+> This section is non-normative.
+
+When working with color on the web, there are a few important terms:
+
+- A *color model* is a mathematical approach to representing colors and their
+  relationships. Historically, RGB has been the dominant color model for both
+  computer monitors and web browsers. Lately, CIELab and OKLab models have
+  shown significant benefits by providing a more *perceptually uniform*
+  distribution of colors, so that similar mathematical adjustments achieve
+  visually similar results.
+- A *color space* is the result of projecting a color model into a coordinate
```

Comment:
Those definitions look weird to me. What you call a "color model" here is exactly what the CSS spec calls a "color space". Color spaces are `sRGB`, `CIE Lab` and `OKLab`, not `RGB` or `HSL`. What you call a color space here is called a coordinate system of the color space in the CSS spec.
I think we should align the terminology to avoid confusion.

Reply:
Reading more in details, I think there is a confusion in the CSS spec between the terminology defined at the beginning of the spec and the the `<color-space>` production in the defined syntax.

Reply:
Yeah, I can try to make some clarifications here, but I think we're technically aligned with the spec already. 

To some extent it's just a problem of confusing and overlapping terms. OKLab & RGB are both color models, and also color spaces (which means they are also coordinate systems). The CSS spec doesn't define _color model_ as a term, they only reference it in the color space term definition. I think the CSS spec is also a bit inconsistent about when they say 'color space' and when they say 'coordinate system'. 

Both of us are also blurring 'color space' with 'color function' a bit, since they're _mostly_ one-to-one, and defining them that way is easier to explain to authors. But that gets a bit tangled, since `rgb` and `srgb` are technically the same space, just using legacy and non-legacy functions. There's no reason for CSS to provide both as part of a `<color-space>` production for 'color interpolation method', but it is useful for us to distinguish between them as a way of tracking legacy vs non-legacy colors. By keeping `rgb` as a distinct space, we no longer need the `is-legacy` boolean.

The other confusion here is that we used non-overlapping examples for an overlapping concept. In my definition, I used legacy function examples for color spaces (rgb and hsl). In their definition examples, they used only color spaces in the `color()` function. Both are accurate. Both `srgb` and `hsl` are color spaces.